packages/core/src/ctype/CType.ts

```diff
@@ -38,18 +33,13 @@ import {
  * @returns CtypeSchema without the $id property.
  */
 export function getSchemaPropertiesForHash(
-  ctypeSchema: CTypeSchemaWithoutId | ICType['schema']
-): Partial<ICType['schema']> {
+  ctypeSchema: ICType | ICTypeSchema
+): ICTypeSchema {
   // We need to remove the CType ID from the CType before storing it on the blockchain
   // otherwise the resulting hash will be different, as the hash on chain would contain the CType ID,
   // which is itself a hash of the CType schema.
-  const schemaWithoutId: Partial<ICType['schema']> =
-    '$id' in ctypeSchema
-      ? (ctypeSchema as ICType['schema'])
-      : (ctypeSchema as CTypeSchemaWithoutId)
-  const shallowCopy = { ...schemaWithoutId }
-  delete shallowCopy.$id
-  return shallowCopy
+  const { $id: _, ...schemaWithoutId } = ctypeSchema as ICType
```

Comment:
Would that simpler version work?
```suggestion
  const { $id, ...schemaWithoutId } = ctypeSchema as ICType
```

Reply:
would also work